pkg/service/FluxService/helper.go

```diff
@@ -0,0 +1,363 @@
+			metadataCompact.Version = version
+		} else if err != nil {
+			return nil, err
+		}
+		fluxManagedResourcesMap[metadataCompact.Id] = metadataCompact.Version
+	}
+	return fluxManagedResourcesMap, nil
+}
+func getReleaseNameNamespace(obj map[string]interface{}, name string) (string, string, error) {
+	//storageNamespace is optional, if not found, then its value byb default is the same as the namespace of helmRelease
+	var storageNamespace string
+
+	if statusRawObj, ok := obj[STATUS]; ok {
+		statusObj := statusRawObj.(map[string]interface{})
+		if storageNamespaceRaw, ok6 := statusObj["storageNamespace"]; ok6 {
+			storageNamespace = storageNamespaceRaw.(string)
+		} else {
+			metadata, found, err := unstructured.NestedMap(obj, "metadata")
+			if err != nil {
+				return "", "", err
+			}
+			if !found {
+				return "", "", fmt.Errorf("entries not found")
+			}
+			namespaceRaw, found, err := unstructured.NestedFieldNoCopy(metadata, "namespace")
+			if err != nil {
+				return "", "", err
+			}
+			if !found {
+				return "", "", fmt.Errorf("entries not found")
+			}
+			storageNamespace = namespaceRaw.(string)
+		}
+	}
+
+	//releaseName is also optional, but if not provided then releaseName is decided by combination of "<targetNamespace>-name" and targetNamespace is optional,in absence of optional targetNamespace, releaseName will be same as HelmRelease.
+
+	spec, found, err := unstructured.NestedMap(obj, "spec")
+	if err != nil {
+		return "", "", err
+	}
+	if !found {
+		return "", "", fmt.Errorf("entries not found")
+	}
+
+	releaseNameRaw, found, err := unstructured.NestedFieldCopy(spec, "releaseName")
+	if err != nil {
+		return "", "", err
+	}
+	if found {
+		return releaseNameRaw.(string), storageNamespace, nil
+	}
+
+	targetNamespaceRaw, found, err := unstructured.NestedFieldNoCopy(spec, "targetNamespace")
+	if err != nil {
+		return "", "", err
+	}
+	if found {
+		return targetNamespaceRaw.(string) + "-" + name, storageNamespace, nil
+	}
+
+	return name, storageNamespace, nil
+
+}
+func getKsAppStatus(obj map[string]interface{}, gvk schema.GroupVersionKind) (*FluxAppStatusDetail, error) {
+	var status, reason, message string
+	if statusRawObj, ok := obj[STATUS]; ok {
+		statusObj := statusRawObj.(map[string]interface{})
+		conditions, found, err := unstructured.NestedSlice(statusObj, "conditions")
+		if err != nil {
+			return nil, err
+		}
+
+		// In case of HelmRelease, Conditions are having negative polarity (means only available when the status is true)
+		if !found && gvk.Kind == FluxAppHelmreleaseKind {
+			status = "Not true"
+			return &FluxAppStatusDetail{
+				Status:  status,
+				Reason:  "StatusNotReady",
+				Message: "Status is not true for this helmRelease",
+			}, nil
+		} else if !found {
+			return nil, fmt.Errorf("conditions not found for the requested  flux app")
+		}
+
+		lastIndex := len(conditions)
+		itemRawObj := conditions[lastIndex-1]
```

Comment:
if conditions key is found and if len of conditions is 0 then this could panic, hence put a length check here